src/platform/silabs/wifi/wf200/WifiInterfaceImpl.cpp

```diff
@@ -929,16 +578,352 @@ static void sl_wfx_generic_status_callback(sl_wfx_generic_ind_t * frame)
+    info.mcast_tx_count    = counters->body.count_tx_multicast_frames;
+    info.ucast_rx_count    = counters->body.count_rx_packets;
+    info.ucast_tx_count    = counters->body.count_tx_packets;
+    info.overrun_count     = gOverrunCount;
+
+    return CHIP_NO_ERROR;
+}
+
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+CHIP_ERROR WifiInterfaceImpl::ConfigurePowerSave()
+{
+    // TODO: Implement Power save configuration. We do a silent failure to avoid causing problems in higher layers.
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR WifiInterfaceImpl::ConfigureBroadcastFilter(bool enableBroadcastFilter)
+{
+    // TODO: Implement Broadcast filtering. We do a silent failure to avoid causing problems in higher layers.
+    return CHIP_NO_ERROR;
+}
+#endif // CHIP_CONFIG_ENABLE_ICD_SERVER
+
+CHIP_ERROR WifiInterfaceImpl::ResetCounters()
+{
+    // TODO: Implement the function
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+}
+
+void WifiInterfaceImpl::ClearWifiCredentials()
+{
+    wifi_provision.Clear();
+}
+
+CHIP_ERROR WifiInterfaceImpl::GetWifiCredentials(WifiCredentials & credentials)
+{
+    VerifyOrReturnError(IsWifiProvisioned(), CHIP_ERROR_INCORRECT_STATE);
+    credentials = wifi_provision;
+
+    return CHIP_NO_ERROR;
+}
+
+bool WifiInterfaceImpl::IsWifiProvisioned()
+{
+    // TODO: We need a better way of checking if we are provisioned or not
+    return wifi_provision.ssid[0] != 0;
+}
+
+void WifiInterfaceImpl::SetWifiCredentials(const WifiCredentials & credentials)
+{
+    wifi_provision = credentials;
+}
+
+CHIP_ERROR WifiInterfaceImpl::ConnectToAccessPoint(void)
+{
+    sl_wfx_security_mode_t connect_security_mode;
+
+    VerifyOrReturnError(IsWifiProvisioned(), CHIP_ERROR_INCORRECT_STATE);
+    ChipLogDetail(DeviceLayer, "WIFI:JOIN to %s", wifi_provision.ssid);
+
+    ChipLogDetail(DeviceLayer,
+                  "WIFI Scan Paramter set to Active channel time %d, Passive Channel "
+                  "Time: %d, Number of prob: %d",
+                  ACTIVE_CHANNEL_TIME, PASSIVE_CHANNEL_TIME, NUM_PROBE_REQUEST);
+    (void) sl_wfx_set_scan_parameters(ACTIVE_CHANNEL_TIME, PASSIVE_CHANNEL_TIME, NUM_PROBE_REQUEST);
+    switch (wifi_provision.security)
+    {
+    case WFX_SEC_WEP:
+        connect_security_mode = sl_wfx_security_mode_e::WFM_SECURITY_MODE_WEP;
+        break;
+    case WFX_SEC_WPA:
+    case WFX_SEC_WPA2:
+        connect_security_mode = sl_wfx_security_mode_e::WFM_SECURITY_MODE_WPA2_WPA1_PSK;
+        break;
+    case WFX_SEC_WPA3:
+        connect_security_mode = sl_wfx_security_mode_e::WFM_SECURITY_MODE_WPA3_SAE;
+        break;
+    case WFX_SEC_NONE:
+        connect_security_mode = sl_wfx_security_mode_e::WFM_SECURITY_MODE_OPEN;
+        break;
+    default:
+        ChipLogError(DeviceLayer, "error: unknown security type.");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    VerifyOrReturnError(sl_wfx_send_join_command(wifi_provision.ssid, wifi_provision.ssidLength, NULL, CHANNEL_0,
```

Comment:
can we add a fix for open access point?

Reply:
I'll create a follow up ticket for this.